src/player.cpp

```diff
@@ -12942,13 +12942,9 @@ void player::assign_activity( const player_activity &act, bool allow_resume )
     activity.warned_of_proximity = false;
 }
 
-bool player::has_activity(const activity_type type) const
+bool player::has_activity(const string_id<activity_type> type) const
```

Comment:
Would be better to create an alias like `using activity_id = string_id<activity_type>;` in the header that declares activities.